hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueue.kt

```diff
@@ -1,14 +1,15 @@
 package uk.gov.justice.hmpps.sqs
 
-import com.amazonaws.services.sqs.AmazonSQS
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlRequest
 
 class HmppsQueue(
   val id: String,
-  val sqsClient: AmazonSQS,
+  val sqsClient: SqsClient,
   val queueName: String,
-  val sqsDlqClient: AmazonSQS? = null,
+  val sqsDlqClient: SqsClient? = null,
   val dlqName: String? = null
 ) {
-  val queueUrl: String by lazy { sqsClient.getQueueUrl(queueName).queueUrl }
-  val dlqUrl by lazy { sqsDlqClient?.getQueueUrl(dlqName)?.queueUrl }
+  val queueUrl: String by lazy { sqsClient.getQueueUrl(GetQueueUrlRequest.builder().queueName(queueName).build()).queueUrl() }
+  val dlqUrl: String? by lazy { sqsDlqClient?.getQueueUrl(GetQueueUrlRequest.builder().queueName(dlqName).build())?.queueUrl() }
 }
```

Comment:
This typifies much of this PR - use the new client `SqsClient` and switch all method calls to the new immutable requests/responses which require builders to create.